azure-common/src/main/java/com/azure/common/http/rest/RestPagedResponseBase.java

```diff
@@ -0,0 +1,77 @@
+package com.azure.common.http.rest;
+
+import com.azure.common.http.HttpHeaders;
+import com.azure.common.http.HttpRequest;
+
+import java.util.List;
+
+public class RestPagedResponseBase<H, T extends Page<T>> implements RestPagedResponse<T> {
+    private final HttpRequest request;
+    private final int statusCode;
+    private final H deserializedHeaders;
+    private final HttpHeaders headers;
+    private final List<T> items;
+    private final String nextLink;
+
+    public RestPagedResponseBase(HttpRequest request, int statusCode, HttpHeaders headers, T body, H deserializedHeaders) {
+        this.request = request;
+        this.statusCode = statusCode;
+        this.headers = headers;
+        this.items = body.items();
+        this.nextLink = body.nextLink();
+        this.deserializedHeaders = deserializedHeaders;
+    }
```

Comment:
Is this constructor necessary? I don't think I see it being used?

Reply:
Ahh. This one is used in the reflection code when you figure out when to use 3 - 5 parameters: https://github.com/Azure/autorest-clientruntime-for-java/blob/a305b748094370d7fdf719deae3956db3e632343/azure-common/src/main/java/com/azure/common/implementation/RestProxy.java#L438-L439

Reply:
How does the nextLink get set in that case?

Reply:
I added a constraint that the class T needs to implement Page, which has nextLink(). And the object is deserialised already when it calls that constructor.